storage/replica_raftstorage.go

```diff
@@ -30,16 +30,24 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
+// Most calls to raft.RawNode require that the replica read lock is held.
```

Comment:
These are not calls to `raft.RawNode`, these are methods of `Replica` that may be called by `raft.RawNode`. This is why we must hold the lock when calling any method of `RawNode`.